src/averageAges.js

```diff
@@ -55,10 +72,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeopleByMother = people.filter(
+    ({ sex, mother: motherName }) => {
+      const mother = people.find(({ name }) => motherName === name);
+
+      return onlyWithSon ? mother && sex === 'm' : mother;
+    });
+
+  return filteredPeopleByMother.reduce(
+    (total, { born, mother }) =>
+      total + (born - people.find(({ name }) => name === mother).born), 0)
+    / filteredPeopleByMother.length;
```

Comment:
Let's simplify this. Create variable for `filteredPeopleByMother.reduce(...)`.
Create variable for `people.find(({ name }) => name === mother).born` inside `reduce`